group-actors/src/main/java/org/sunbird/service/impl/GroupServiceImpl.java

```diff
@@ -82,15 +85,39 @@ public String createGroup(Group groupObj) throws BaseException {
                 memberService.fetchMembersByGroupIds(Lists.newArrayList(groupId), null);
             dbResGroup.put(JsonKey.MEMBERS, members);
           }
-        }
-        if (CollectionUtils.isEmpty(fields) || !fields.contains(JsonKey.ACTIVITIES)) {
-          dbResGroup.remove(JsonKey.ACTIVITIES);
+          if (fields.contains(JsonKey.ACTIVITIES)) {
+            List<Map<String, Object>> dbResActivities =
+                (List<Map<String, Object>>) dbResGroup.get(JsonKey.ACTIVITIES);
+            if (dbResActivities != null && !dbResActivities.isEmpty()) {
+              addActivityInfoDetails(dbResActivities);
+            }
+          } else {
+            removeActivities(dbResGroup);
+          }
+        } else {
+          removeActivities(dbResGroup);
         }
       }
     }
     return dbResGroup;
   }
 
+  private Object removeActivities(Map<String, Object> dbResGroup) {
+    return dbResGroup.remove(JsonKey.ACTIVITIES);
+  }
+
+  private void addActivityInfoDetails(List<Map<String, Object>> dbResActivities) {
+    Map<String, Map<String, Object>> activityInfoMap =
+        searchActivityService.searchActivity(dbResActivities);
+    for (Map<String, Object> activity : dbResActivities) {
+      if (activityInfoMap.containsKey(activity.get(JsonKey.ID))) {
+        activity.put(JsonKey.ACTIVITY_INFO, activityInfoMap.get(activity.get(JsonKey.ID)));
+      } else {
+        activity.put(JsonKey.ACTIVITY_INFO, new HashMap<>());
```

Comment:
@sknirmalkar89 - why do we need to send an empty map if activity info is not retrieved? we can just skip the activity info section for activities whose data is not retrieved.